modules/1_opencv/src/opencv_coins.cpp

```diff
@@ -1,14 +1,47 @@
 #include "opencv_coins.hpp"
-
+#include<math.h>
 using namespace cv;
 
 unsigned countCoins(const Mat& img) {
-    Mat gray, thresh;
+    Mat gray, thresh, sure_bg,  sure_fg;
     cvtColor(img, gray, COLOR_BGR2GRAY);
     threshold(gray, thresh, 0, 255, THRESH_BINARY_INV + THRESH_OTSU);
     // Remove unfilled black holes by morphological closing
     morphologyEx(thresh, thresh, MORPH_CLOSE, Mat::ones(3, 3, CV_8U), Point(-1, -1), 3);
+    //sure background area
+    sure_bg = thresh;
+    dilate(thresh, sure_bg,Mat::ones(3, 3, CV_8U), Point(-1, -1), 3);
+    // Finding sure foreground area
+    erode(thresh, sure_fg, Mat::ones(3, 3, CV_8U), Point(-1, -1), 5);
+    distanceTransform(sure_fg, sure_fg, DIST_L2, 5);
+    double min_thresh, max_threh;
+    minMaxLoc(sure_fg, &min_thresh, &max_threh);
+    threshold(sure_fg, sure_fg,0.7 * max_threh,255, THRESH_BINARY);
+    sure_fg.convertTo(sure_fg, CV_8U, 1, 0);
+    imshow("ex_4", sure_fg);
```

Comment:
Why do you use `imshow` this?

Reply:
I wanted to display the image on the screen to see what happened to the image after applying the treshold method.